### ISSUE TITLE
Add Android "App shortcuts"

### ISSUE THREAD

**Issue by Sosasees** [godotengine/godot-proposals#2739]
### Describe the project you are working on

I am working on an Android game with a title screen.

### Describe the problem or limitation you are having in your project

I want to implement an "App shortcut"
so that the player can select and continue playing their latest Save File already, bypassing the title screen and main menu completely.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

![image](https://user-images.githubusercontent.com/66305550/118689832-b4a62000-b807-11eb-8db6-3b0e3a25152a.png)
When holding down an icon of an app that supports "app shortcuts" on the home screen, it'll open a window with these "app shortcuts". These are shortcuts to other parts of the app that aren't the app's normal starting screen.

This is where I want to put my second "Continue" button that's outside of the game instead of its title screen.

[This](https://developer.android.com/guide/topics/ui/shortcuts/) is the Android Developer's documentation for App Shortcuts.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Shortcut types:
```
enum AndroidAppShortcuts.ShortcutType {
ANDROID_APP_SHORTCUT_TYPE_STATIC,
ANDROID_APP_SHORTCUT_TYPE_DYNAMIC,
ANDROID_APP_SHORTCUT_TYPE_PINNED }
```

To add shortcut:
```
if OS.name == "Android":
	AndroidAppShortcuts.add_shortcut(shortcut_id, shortcut_type, "path://to/scene", "Shortcut Description", "path://to/scene")
```

To update shortcut:
```
if OS.name == "Android":
	AndroidAppShortcuts.update_shortcut(shortcut_id, shortcut_type, "path://to/scene", "Shortcut Description", "path://to/scene")
```

To disable shortcut:
```
if OS.name == "Android":
	AndroidAppShortcuts.disable_shortcut(shortcut_id)
```

To enable shortcut:
```
if OS.name == "Android":
	AndroidAppShortcuts.enable_shortcut(shortcut_id)
```

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I haven't found anything.

### Is there a reason why this should be core and not an add-on in the asset library?

If possible, it might actually be better as an add-on instead of a core feature.

**Comment by hani09876** [comment]
Did they add this feature

**Comment by Calinou** [comment]
@hani09876 No, not yet. There is no ETA for implementing this feature, since nobody has looked into implementing it yet.

**Comment by DawidMoza** [comment]
I recently created a plugin for app shortcuts, it currently only supports creating feedback mail, but there is a possibility to expand it in the future for more general use: https://github.com/DawidMoza/quick-actions-app-shortcuts

You can submit an issue describing what exactly do You need

**Comment by poyribn** [comment]
The App could set a Value in the Project&nbsp;Settings for the App&nbsp;Shortcut that was used to start the current&nbsp;Instance — including ``null`` if the App launched normally without using an App&nbsp;Shortcut.
Then i could adjust the App’s Behavior by querying that Project&nbsp;Setting using [``if``&nbsp;Statements](https://docs.godotengine.org/en/stable/classes/class_dictionary.html#dictionary).

I might also need the Ability to change the App&nbsp;Shortcuts with&nbsp;GDScript. For&nbsp;Example:

- If i use App&nbsp;Shortcuts to link to saved&nbsp;Games, changing them with GDScript is useful to show the Name and Profile&nbsp;Picture of each saved&nbsp;Game.
- If i use App&nbsp;Shortcuts for Extra&nbsp;Features instead, i could add App&nbsp;Shortcuts for each Extra when it’s unlocked and&nbsp;not&nbsp;earlier.

And finally, my Game should have the Ability to respond to App&nbsp;Shortcuts that were removed or not&nbsp;created&nbsp;yet
because they could still exist on the User’s Android&nbsp;Home&nbsp;Screen. For&nbsp;Example:

- If i import a Home&nbsp;Screen&nbsp;Layout with the Radio&nbsp;Extra from another&nbsp;Phone
but i have not unlocked the Radio on the new&nbsp;Phone&nbsp;yet,
the Game should still respond as&nbsp;expected to the ``"extra_radio"`` StringName and unlock the&nbsp;Radio.
- If i try to start a nonexisting saved&nbsp;Game, or a saved&nbsp;Game whiches Name in the Shortcut doesn’t match its actual&nbsp;Name, i should script something to handle these Errors&nbsp;gracefully.
To make this possible, the App&nbsp;Shortcut would set the App&nbsp;Shortcut Value to a [Dictionary](https://docs.godotengine.org/en/stable/classes/class_dictionary.html#dictionary) or a custom&nbsp;Type&nbsp;Resource with the following Variables: ``StringName display_name`` and ``String save_id``
